src/Filter/SearchFilter.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types = 1);
+
+namespace PommProject\ApiPlatform\Filter;
+
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use PommProject\Foundation\Where;
+
+/**
+ * @author Mikael Paris <stood86@gmail.com>
+ */
+final class SearchFilter implements FilterInterface
+{
+    /**
+     * @var string Exact matching
+     */
+    const STRATEGY_EXACT = 'exact';
+
+    /**
+     * @var string The value must be contained in the field
+     */
+    const STRATEGY_PARTIAL = 'partial';
+
+    /**
+     * @var string Finds fields that are starting with the value
+     */
+    const STRATEGY_START = 'start';
+
+    /**
+     * @var string Finds fields that are ending with the value
+     */
+    const STRATEGY_END = 'end';
+
+    /**
+     * @var string Finds fields that are starting with the word
+     */
+    const STRATEGY_WORD_START = 'word_start';
+
+    public function andWhereByStrategy(string $strategy = null, Where $where, string $property, string $value): Where
+    {
+        $operator = 'like';
+        $caseSensitive = true;
+        if ($strategy !== null && 0 === strpos($strategy, 'i')) {
+            $strategy = substr($strategy, 1);
+            $operator = 'ilike';
+            $caseSensitive = false;
+        }
+
+        switch ($strategy) {
+            case null:
```

Comment:
Instead of dealing with `null` value in multiple places, I prefer setting `$strategy` early with the default value:

```php
if ($strategy === null) {
    $strategy = self::STRATEGY_EXACT;
}
```
Thus you can also simplify the condition line 43.